### PR TITLE
Replacing sssd updates to use IniParse

### PR DESCRIPTION
**Pull request by abellotti**
Preferring to use IniParse vs. config[] way
- Simpler code
- Easier to understand/update
- Could be leveraged for AD/Ldap

Removed the need for the class instance timestamp variable.
- common config_file_backup method.

_Note_: viewing in split mode shows better the simplification.

### CODE REVIEW

**Comment by abellotti** [comment]
code climate above issue ok,  the AD specific configure_sssd will be changing in an upcoming PR.

**Comment by abellotti** [comment]
solves #4

File: lib/httpd_configmap_generator/base/sssd.rb [hunk 1]
```diff
@@ -1,58 +1,55 @@
...
-          config[/\[ifp\](\n.*)+user_attributes = (.*)/, 2] = user_attributes
-        else
-          config[/\[ifp\](\n)/, 1] = ifp_config
-        end
-      else
-        config << "\n[ifp]#{ifp_config}\n"
+    private
+
+    def section(key)
+      if key == "domain"
```
Comment (jvlcek) [inline lib/httpd_configmap_generator/base/sssd.rb]: @abellotti I think we can eliminate these 2 checks and only user
`if key =~ /^domain\/.* $/`

**Comment by abellotti** [comment]
Thanks @jvlcek and @gtanzillo will rebase the next PR now.

**Pull request merged**